cstree/src/syntax/iter.rs

```diff
@@ -166,3 +184,37 @@ impl<S: Syntax, D> ExactSizeIterator for SyntaxElementChildren<'_, S, D> {
     }
 }
 impl<S: Syntax, D> FusedIterator for SyntaxElementChildren<'_, S, D> {}
+
+#[cfg(test)]
+#[allow(dead_code)]
+mod tests {
+    use std::marker::PhantomData;
+
+    use super::*;
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    struct DummyKind;
+
+    impl Syntax for DummyKind {
+        fn from_raw(_: crate::RawSyntaxKind) -> Self {
+            unreachable!()
+        }
+
+        fn into_raw(self) -> crate::RawSyntaxKind {
+            unreachable!()
+        }
+
+        fn static_text(self) -> Option<&'static str> {
+            unreachable!()
+        }
+    }
+
+    struct NotClone;
+
+    fn assert_clone<C: Clone>(_: PhantomData<C>) {}
+
+    fn test_impls_clone() {
+        assert_clone(PhantomData::<SyntaxNodeChildren<DummyKind, NotClone>>);
+        assert_clone(PhantomData::<SyntaxElementChildren<DummyKind, NotClone>>);
+    }
```

Comment:
I believe this can be slightly simplified to

```suggestion
    fn assert_clone<C: Clone>() {}

    fn test_impls_clone() {
        assert_clone::<SyntaxNodeChildren<DummyKind, NotClone>>();
        assert_clone::<SyntaxElementChildren<DummyKind, NotClone>>();
    }
```